TechAndCheck/zenodotus
# Revert back to CSS/JS bundling from importmaps

_The background to this issue has been discussed ad nauseum in issues like #188 and on Slack, so I won't go into the full justification here._

This PR removes our use of `importmap-rails` and goes back to the boring old days of asset-bundling via a JS bundler, though in this case [esbuild](https://esbuild.github.io/) rather than Webpack because it's light and fast.

The commits are organized pretty atomically, so I recommend poking through them individually if you're curious about individual changes. fe06b3779f9c9f10143baf022fb9e96e70707be4 was huge by necessity, but I put a lot of clarifying detail into its commit message.

**Testing:**
- Remove `node_modules/`
- Empty `app/assets/builds/` and `public/assets/` to make sure old assets aren't hanging around
- `yarn install`
- `bundle install`
- Run and test the app using one of these two methods:
   - `./bin/dev` will fire up the CSS/JS bundler alongside Puma, so no more work needed; **this is preferred for testing**
   - `rails s` will need `rails assets:precompile` to be run first; then all will be well, though CSS/JS changes won't compiled automatically (I don't think)
